axum/src/response/sse.rs

```diff
@@ -174,6 +176,23 @@ pub struct Event {
     flags: EventFlags,
 }
 
+#[derive(Debug)]
+#[must_use]
+/// Expose [`Event`] as a [`std::io::Write`]
+/// such that any form of data can be written as data safely.
+///
+/// This also ensures that newline characters `\r` and `\n`
+/// correctly trigger a split with a new `data: ` prefix.
+///
+/// # Panics
+///
+/// Panics if any `data` has already been written prior to the first write
+/// of this [`EventDataWrite`] instance.
+pub struct EventDataWrite<'a> {
```

Comment:
By analogy of the naming guideline [C-ITER-TY](https://rust-lang.github.io/api-guidelines/naming.html#iterator-type-names-match-the-methods-that-produce-them-c-iter-ty), it might make more sense to call this struct `EventDataWriter`, since it is returned by the method `Event::data_writer`. Also, this would be inline with e.g. `BufWriter`.